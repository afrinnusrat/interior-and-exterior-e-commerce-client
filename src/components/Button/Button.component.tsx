import React from 'react';

import { ButtonWrapper } from './Button.styles';

interface ButtonProps {
  disabled?: boolean;
  onClick?: () => void;
  btnType?: string;
  type?: 'button' | 'submit' | 'reset' | undefined;
  children: any;
}

export const Button: React.FC<ButtonProps> = ({
  disabled,
  onClick,
  btnType,
  type = 'button',
  children
}) => (
  <ButtonWrapper
    data-testid="ButtonWrapper"
    disabled={disabled}
    onClick={onClick}
    className={[btnType].join(' ')}
    type={type}
  >
    {children}
  </ButtonWrapper>
);
