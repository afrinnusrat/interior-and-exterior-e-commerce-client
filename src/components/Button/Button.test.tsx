import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Button } from './Button.component';

afterEach(cleanup);

describe('<Button />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Button>Submit</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains disabled attribute', () => {
    const { getByTestId } = render(<Button disabled={true}>Submit</Button>);
    expect(getByTestId('ButtonWrapper')).toHaveAttribute('disabled', '');
  });

  it('calls `onClick` function', () => {
    const mockOnClick = jest.fn();
    const { getByTestId } = render(
      <Button onClick={mockOnClick}>Submit</Button>
    );
    fireEvent.click(getByTestId('ButtonWrapper'));
    expect(mockOnClick).toBeCalledTimes(1);
  });

  it('contains `dark` class', () => {
    const { getByTestId } = render(<Button btnType="dark">Submit</Button>);
    expect(getByTestId('ButtonWrapper')).toHaveClass('dark');
  });

  it('contains `submit` tyle', () => {
    const { getByTestId } = render(<Button type="submit">Submit</Button>);
    expect(getByTestId('ButtonWrapper')).toHaveAttribute('type', 'submit');
  });

  it('contains `Submit` text', () => {
    const { getByTestId } = render(<Button>Submit</Button>);
    expect(getByTestId('ButtonWrapper')).toHaveTextContent('Submit');
  });
});
