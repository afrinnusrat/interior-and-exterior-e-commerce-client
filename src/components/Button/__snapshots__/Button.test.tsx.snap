// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa cPICdt"
    data-testid="ButtonWrapper"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
